src/Microsoft.VisualStudio.AppDesigner/PropPageDesigner/PropertyPageSerializationService_Store.vb

```diff
@@ -131,7 +132,9 @@ Namespace Microsoft.VisualStudio.Editors.PropPageDesigner
             ''' <param name="Stream">The stream to load from</param>
             Public Shared Function Load(Stream As Stream) As PropertyPageSerializationStore
                 Dim f As New BinaryFormatter
-                Return DirectCast(f.Deserialize(Stream), PropertyPageSerializationStore)
+                Dim StreamObject As Object = f.Deserialize(Stream)
+                TelemetryLogger.LogBinaryFormatterEvent(NameOf(Load), NameOf(PropertyPageSerializationStore), TelemetryLogger.BinaryFormatterType.Deserialize)
+                Return DirectCast(StreamObject, PropertyPageSerializationStore)
```

Comment:
I see throughout this PR that you're deserialising before logging telemetry. Is there a reason for that? Logging first would make it a smaller diff.

Reply:
Logging after using the binary formatter also means that a failure will not result in a telemetry event (even though I think you would expect one)

Reply:
I don't know the standard practice for telemetry. I'm used to debug logging, in which, you'd log an event after it occurs, since the log entry indicates it *has* happened. If you log prior, you can't actually determine if the operation happened or not. Since this particular set of data doesn't have a strong relationship to the operation itself, I can log preemptively. I only really care if the function itself was called at this moment.

Reply:
Updated. Also normalized BinaryFormatter calling.